src/brod_client.erl

```diff
@@ -186,8 +189,9 @@ handle_info({'EXIT', Pid, Reason}, State) ->
 handle_info(_Info, State) ->
   {noreply, State}.
 
-handle_call({get_topic_worker, Topic}, _From, State) ->
-  Result = do_get_topic_worker(State, Topic),
+handle_call({find_producer, Topic, Partition}, _From, State) ->
+  SupPid = State#state.producers_sup,
+  Result = brod_producers_sup:find_producer(SupPid, Topic, Partition),
```

Comment:
no, you can't do this in a handle_call, causing dead lock. (same reason why i implemented get_topic_worker before).
here you should return the second-level supervisor pid as quick as possible and then call brod_supervisor:find_child(Partition) in the caller process instead.

Reason:
brod_supervisor:find_child is a gen_server:call to SUP2, but in SUP2's post_init handle_info, it'll make a call to brod_client to get metadata, brod_client might be currently blocked by this find_producer call.